src/client/clientlauncher.cpp

```diff
@@ -35,6 +35,10 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "renderingengine.h"
 #include "network/networkexceptions.h"
 
+#if USE_DISCORD
```

Comment:
I *really* dislike the idea of having discord-pecific code scattered everywhere. I would suggest instead making a "rich presence" or "game status" interface, which would be a superset of what discord offers. Then, you use that unconditionally, and it decides itself whether to use discord APIs or not. That way, it could also be reused easily (say, with the steam API or other).